src/components/PopupWithForm.js

```diff
@@ -27,8 +40,11 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this.close();
+      this.handleLoading(true);
+      this._handleFormSubmit(this._getInputValues())
+        .then(() => {
+          this.close();
+        })
```

Comment:
* You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.



```js
      this._handleFormSubmit(this._getInputValues())
        .then(() => {
          this.close();
        })
        .finally(() => {  // here
          this.handleLoading(false);
        });

```